nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/AttributesToJSON.java

```diff
@@ -59,9 +60,37 @@
 @Tags({"json", "attributes", "flowfile"})
 @InputRequirement(InputRequirement.Requirement.INPUT_REQUIRED)
 @CapabilityDescription("Generates a JSON representation of the input FlowFile Attributes. The resulting JSON " +
-        "can be written to either a new Attribute 'JSONAttributes' or written to the FlowFile as content.")
+        "can be written to either a new Attribute 'JSONAttributes' or written to the FlowFile as content. Attributes " +
+        " which contain nested JSON objects can either be handled as JSON or as escaped JSON depending on the strategy chosen.")
 @WritesAttribute(attribute = "JSONAttributes", description = "JSON representation of Attributes")
 public class AttributesToJSON extends AbstractProcessor {
+    public enum JsonHandlingStrategy implements DescribedValue {
+        ESCAPED_STRING("Escapes any nested JSON as a string", "Escaped String"),
+        NESTED_OBJECT("Handles nested JSON as JSON", "Nested Object");
+
+        private final String description;
+        private final String displayName;
+
+        JsonHandlingStrategy(String description, String displayName) {
```

Comment:
As a general convention, recommend placing the display name before the description.
```suggestion
        ESCAPED_STRING("Escaped String", "Escapes JSON attribute values to strings"),
        NESTED_OBJECT("Nested Object", "Handles JSON attribute values as nested structured objects");

        private final String description;
        private final String displayName;

        JsonHandlingStrategy(String displayName, String description) {
```